### PR TITLE
[ModelRunner] apply_grammer uses vllm function

### PR DESCRIPTION
**Pull request by zhenwenqi2024**
### What this PR does / why we need it?
this pr removes apply_gramme in npu_model_runner. we change logits to cpu, and do the same thing with gpu_model_runner.
it may change the performance, we will change it after torch.compile is supported with npu inductor

### How was this patch tested?

- vLLM version: v0.12.0
- vLLM main: https://github.com/vllm-project/vllm/commit/ad32e3e19ccf0526cb6744a5fed09a138a5fb2f9

### CODE REVIEW

**Review by MengqingCao — APPROVED** [review]
LGTM

**Comment by wangxiyuan** [comment]
Please update commit message

**Comment by wangxiyuan** [comment]
please fix the merge conflict, I'll merge it directly.
